rotkehlchen/tests/unit/test_evm_contracts.py

```diff
@@ -33,3 +64,71 @@ def test_evm_abi_data(globaldb):
+        assert ethereum_inquirer.contracts.abi(TEST_ABI_NAME) == TEST_ABI  # type: ignore[call-overload]  # noqa: E501
+
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        abi_in_db = cursor.execute(
+            'SELECT value FROM contract_abi WHERE name=?',
+            (TEST_ABI_NAME,),
+        ).fetchone()[0]
+        assert json.loads(abi_in_db) == TEST_ABI
+
+
+@pytest.mark.parametrize('reuse_abi', [True, False])
+def test_get_remote_contract(ethereum_inquirer: 'EthereumInquirer', reuse_abi: bool):
+    """
+    Tests that pulling an evm contract from rotki data repository works.
+    If reuse_abi is True, then checks that if an abi already exists for the contract, it is reused.
+    If reuse_abi is False then makes sure that a new abi entry is added as the result of pulling.
+    """
+    if reuse_abi is True:
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            write_cursor.execute(
+                'INSERT INTO contract_abi(name, value) VALUES(?, ?)',
+                (TEST_CONTRACT['abi_name'], json.dumps(TEST_CONTRACT['abi'], separators=(',', ':'))),  # noqa: E501
+            )
+
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        initial_abi_count = cursor.execute('SELECT COUNT(*) FROM contract_abi').fetchone()[0]
+
+    def requests_mock(url, timeout):
+        if url == f'https://raw.githubusercontent.com/rotki/data/develop/contracts/contract/{TEST_CONTRACT_NAME}.json':  # noqa: E501
+            return MockResponse(status_code=200, text=json.dumps(TEST_CONTRACT))
+
+        raise AssertionError(f'Unexpected url {url} was called')
+
+    expected_contract = EvmContract(
+        address=TEST_CONTRACT['address'],  # type: ignore[arg-type]
+        abi=TEST_CONTRACT['abi'],  # type: ignore[arg-type]
+        deployed_block=TEST_CONTRACT['deployed_block'],  # type: ignore[arg-type]
+    )
+    with patch('requests.get', side_effect=requests_mock):
+        assert ethereum_inquirer.contracts.contract(TEST_CONTRACT_NAME) == expected_contract  # type: ignore[call-overload]  # noqa: E501
+
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        contract_data_in_db = cursor.execute(
+            'SELECT address, chain_id, contract_data.name, contract_abi.value, deployed_block '
+            'FROM  contract_data INNER JOIN contract_abi ON contract_data.abi=contract_abi.id '
+            'WHERE contract_data.name=?',
+            (TEST_CONTRACT_NAME,),
+        ).fetchone()
+        assert contract_data_in_db == (
+            TEST_CONTRACT['address'],
+            TEST_CONTRACT['chain_id'],
+            TEST_CONTRACT_NAME,
+            json.dumps(TEST_CONTRACT['abi'], separators=(',', ':')),
+            TEST_CONTRACT['deployed_block'],
+        )
+        expected_abi_count = initial_abi_count if reuse_abi is True else initial_abi_count + 1
+        assert cursor.execute('SELECT COUNT(*) FROM contract_abi').fetchone()[0] == expected_abi_count  # noqa: E501
```

Comment:
can you also check that querying a contract that exists doesn't trigger a remote call? to prevent that scenario if we make changes in the future